libminifi/include/utils/requirements/HasSpaceshipOperator.h

```diff
@@ -0,0 +1,31 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <type_traits>
+
+namespace org::apache::nifi::minifi::utils {
+
+template<typename T, typename = std::void_t<>>
+struct has_spaceship_operator: std::false_type {};
+
+template<typename T>
+struct has_spaceship_operator<T, std::void_t<decltype(std::declval<T>() <=> std::declval<T>())>> : std::true_type {};
```

Comment:
I managed to accidentally comment on the commit instead of the PR. So here it comes again:

1. I'd keep this in the same file as the usage, as an implementation detail. (`namespace detail`)
2. We should use the concepts language tool instead of legacy metaprogramming hacks.

```cpp
template<typename T>
concept has_spaceship_operator = requires(T t, T u) {
  { t <=> u };
};
```

Reply:
Updated in 96933e8c5206c3dd96f0a794260f41b49cce4d19